### PR TITLE
Simplify Makefile

### LINKED ISSUE #109: Makefile refactoring
Makefile targets are dependent on `full.sh` and environment variables declared at the same file.
 
- refactor Makefile to `GNU call Functions`  to get rid of dependencies on `full.sh`. 
- refactor all places where `full.sh` is used (i.e. terratest build pipe)

### PR DESCRIPTION
**Pull request by kuritka**
close #109 

All maintenance logic is within one `Makefile` now.  

 - Remove dependencies on `full.sh`, `registry.sh`
-  github workflow: Move logic from **terratest** `K8GB deployment` into `Makefile`
- Helm charts:  [new repo location](https://helm.sh/blog/new-location-stable-incubator-charts/) from `https://kubernetes-charts.storage.googleapis.com` to `https://charts.helm.sh/stable`
- revision all Makefile targets and fix if necessary
- optimize publically exposed targets (see report below). Move logic from targets into UDF's
- section highlighting for large scripts `deploy-full-local-setup`, `deploy-full-terratest-setup`
Screenshot 2020-11-11 at 13 27 48
- fix mistaken host-names in documentation 

## Report

- **bold** - moved to new Makefile and accessible `make <target>`
 - ~crossed out~ - functionality moved to new Makefile but not accessible via targets (no reason to expose)

# old
 -  **all** _(kubebuilder initial)_
 - ~bundle~ REMOVED , obsolete by 1.0 migration
 - ~bundle-build~ REMOVED , obsolete by 1.0 migration
 - **clean-test-apps**
 - ~controller-gen~ _(kubebuilder initial)_
 - ~create-k8gb-ns~
 - ~create-test-ns~
 - ~debug-test-local~ -> **debug-idea** new implementation
 - **debug-test-etcd**
 - **demo-failover**
 - **demo-roundrobin**
 - **deploy** _(kubebuilder initial)_
 - ~deploy-first-k8gb~
 - **deploy-full-local-setup**
 - ~deploy-gslb-cr~
 - ~deploy-gslb-operator~
 - **deploy-gslb-operator-14**
 - ~deploy-local-cluster~ REMOVED
 - ~deploy-local-ingress~
 - ~deploy-second-k8gb~
 - ~deploy-test-apps~
 - ~deploy-two-local-clusters~
 - **destroy-full-local-setup**
 - ~destroy-local-cluster~ REMOVED
 - ~destroy-two-local-clusters~
 - **dns-smoke-test**
 - **dns-tools**
 - **docker-build** _(kubebuilder initial)_
 - **docker-push** _(kubebuilder initial)_
 - **docker-test-build-push**
 - ~fmt~ is part of golangci, using lint instead, REMOVED; _(kubebuilder initial)_
 - ~generate~ _(kubebuilder initial)_
 - **infoblox-secret**
 - **init-failover**
 - **init-round-robin**
 - **install** _(kubebuilder initial)_
 - ~kustomize~
 - **lint**
 - **manager** _(kubebuilder initial)_
 - ~manifests~ _(kubebuilder initial)_
 - **reset (NEW)**
 - **run** _(kubebuilder initial)_
 - **start-test-app**
 - **stop-test-app**
 - **terratest**
 - **test** _(kubebuilder initial)_
 - **test-failover**
 - **test-round-robin**
 - **uninstall** *_(kubebuilder initial)_
 - ~use-first-context~
 - ~use-second-context~
 - **version**
 - ~vet~ is part of golangci, using lint instead REMOVED; _(kubebuilder initial)_
 - ~wait-for-gslb-ready~
 - ~wait-for-nginx-ingress-ready~

# new
 - all  _(kubebuilder initial)_
 - clean-test-apps
 - debug-idea
 - debug-test-etcd
 - demo-failover
 - demo-roundrobin
 - deploy-full-local-setup
 - destroy-full-local-setup
 - deploy _(kubebuilder initial)_
 - deploy-gslb-operator-14
 - dns-smoke-test
 - dns-tools
 - docker-build _(kubebuilder initial)_
 - docker-push _(kubebuilder initial)_
 - docker-test-build-push
 - infoblox-secret
 - init-failover
 - init-round-robin
 - install _(kubebuilder initial)_
 - lint
 - **list (NEW)**
 - manager _(kubebuilder initial)_
 - **reset (NEW)**
 - run _(kubebuilder initial)_
 - start-test-app
 - stop-test-app
 - terratest
 - test _(kubebuilder initial)_
 - test-failover
 - test-round-robin
 - uninstall _(kubebuilder initial)_
 - version

### CODE REVIEW

**Review by somaritane — CHANGES_REQUESTED** [review]
@kuritka looks good, few minor comments

File: Makefile [hunk 1]
```diff
@@ -1,267 +1,161 @@
...
-HOST_ALIAS_IP1 ?= 172.17.0.9
-HOST_ALIAS_IP2 ?= 172.17.0.5
-K8GB_IMAGE_REPO ?= absaoss/k8gb
-K8GB_IMAGE_TAG  ?= v$(VERSION)
-K8GB_COREDNS_IP ?= kubectl get svc k8gb-coredns -n k8gb -o custom-columns='IP:spec.clusterIP' --no-headers
-PODINFO_IMAGE_REPO ?= stefanprodan/podinfo
 COMMIT_HASH ?= $(shell git rev-parse --short HEAD)
 
-.PHONY: debug-local
-debug-local: create-test-ns
```
[the remark this answers is not in the document]
Comment (kuritka) [reply Makefile:143]: Hi @somaritane , all comments are reasonable and will amend changes. 


Regarding `debug-local`,  after migration to 1.0.0 `operator-sdk run --local --namespace=test-gslb --enable-delve` generates `FATA[0000] must run command in project root dir: project structure requires build/Dockerfile` which leads to some crazy workarounds. 

I searched for some fancy solution and [this](https://dev4devs.com/2019/05/04/operator-framework-how-to-debug-golang-operator-projects/)  work exactly as expected - unfortunately implementation depends on IDE.  `debug-idea` is already there.  `debug-vs` needs to be implemented.

[the remark this answers is not in the document]
Comment (somaritane) [reply Makefile:143]: @kuritka, thank you for explanation, I'll look into `debug-vscode` later then 👍

**Review by somaritane — APPROVED** [review]
lgtm 👍

**Review by ytsarev — APPROVED** [review]

**Pull request merged**